framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1757,7 +1760,7 @@ public AnnotatedNullType getAnnotatedNullType(Set<? extends AnnotationMirror> an
      * @param tree tree to whose inner most enclosing method or class is returned.
      * @return the inner most enclosing method or class tree of {@code tree}
      */
-    protected Tree getEnclosingClassOrMethod(Tree tree) {
+    protected @Nullable Tree getEnclosingClassOrMethod(Tree tree) {
```

Comment:
Shouldn't all code be within at least a class? Can we specify more clearly when this method can return null?
In the long term, we should probably think about checking for enclosing lambda's and field declarations and how to handle such trees.

Reply:
You are right.  Thanks for catching my goof.  I will rework this.